training/model_management/src/azureml/model/mgmt/processors/pyfunc/vision/detection_predict.py

```diff
@@ -0,0 +1,185 @@
+
+    :param request_body: Image
+    :type request_body: bytes
+    :param parent_dir: directory name
+    :type parent_dir: str
+    :return: Path to the file
+    :rtype: str
+    """
+    with tempfile.NamedTemporaryFile(dir=parent_dir, mode="wb", delete=False) as image_file_fp:
+        # image_file_fp.write(request_body)
+        img_path = image_file_fp.name + ".png"
+        img = Image.open(io.BytesIO(request_body))
+        img.save(img_path)
+        return img_path
+
+
+def _process_image(img: pd.Series) -> pd.Series:
+    """Process the image input (image url or base64 string) and return bytes.
+
+    :param img: pandas series with image in base64 string format or url
+    :type img: pd.Series
+    :return: decoded image in pandas series format.
+    :rtype: Pandas Series
+    """
+    image = img[0]
+    if _is_valid_url(image):
+        image = requests.get(image).content
+        return pd.Series(image)
+    try:
+        return pd.Series(base64.b64decode(image))
+    except Exception:
+        raise ValueError("Invalid image format")
+
+
+def _is_valid_url(text: str) -> bool:
+    """Check if text is url or base64 string.
+
+    :param text: text to validate
+    :type text: str
+    :return: True if url else false
+    :rtype: bool
+    """
+    regex = (
+        "((http|https)://)(www.)?"
+        + "[a-zA-Z0-9@:%._\\+~#?&//=]"
+        + "{2,256}\\.[a-z]"
+        + "{2,6}\\b([-a-zA-Z0-9@:%"
+        + "._\\+~#?&//=]*)"
+    )
+    p = re.compile(regex)
+
+    # If the string is empty
+    # return false
+    if str is None:
+        return False
+
+    # Return if the string
+    # matched the ReGex
+    if re.search(p, text):
+        return True
+    else:
+        return False
+
+
+class ImagesDetectionMLFlowModelWrapper(mlflow.pyfunc.PythonModel):
+    """MLFlow model wrapper for AutoML for Images models."""
+
+    def __init__(
+        self,
+        task_type: str,
+    ) -> None:
+        """Mlflow model wrapper for AutoML for Images models.
+
+        :param task_type: Task type used in training.
+        :type task_type: str
+        """
+        super().__init__()
+        self._model = None
+        self._inference_detector = None
+        self._task_type = task_type
+
+    def load_context(self, context: mlflow.pyfunc.PythonModelContext) -> None:
+        """Load a Mlflow model with pyfunc.load_model().
+
+        :param context: Mlflow context containing artifacts that the model can use for inference
+        :type context: mlflow.pyfunc.PythonModelContext
+        """
+        print("Inside load_context()")
+
+        if self._task_type == Tasks.MM_OBJECT_DETECTION.value:
+            # Install mmcv and mmdet using mim, with pip installation is not working
+            subprocess.check_call([sys.executable, "-m", "mim", "install", "mmcv-full==1.7.1"])
+            subprocess.check_call([sys.executable, "-m", "mim", "install", "mmdet==2.28.2"])
+            subprocess.check_call(
```

Comment:
I would suggest to add logic for why do we need to install opencv like this

Reply:
Thanks. Added.